src/main/java/io/vertx/core/net/impl/TCPServerBase.java

```diff
@@ -96,21 +91,10 @@ public synchronized SSLHelper sslHelper() {
 
   public Future<TCPServerBase> bind(SocketAddress address) {
     ContextInternal listenContext = vertx.getOrCreateContext();
-
-    io.netty.util.concurrent.Future<Channel> bindFuture = listen(address, listenContext);
-
-    Promise<TCPServerBase> promise = listenContext.promise();
-    bindFuture.addListener(res -> {
-      if (res.isSuccess()) {
-        promise.complete(this);
-      } else {
-        promise.fail(res.cause());
-      }
-    });
-    return promise.future();
+    return listen(address, listenContext).map(this);
   }
 
-  private synchronized io.netty.util.concurrent.Future<Channel> listen(SocketAddress localAddress, ContextInternal context) {
+  private synchronized Future<Channel> listen(SocketAddress localAddress, ContextInternal context) {
```

Comment:
I am having issues tracing through this method and understanding the reasoning. Any chance of adding comments or decomposing into smaller methods?

Reply:
I'll add comments